### ISSUE TITLE
The Evolution recurrence pattern is not compatible with Outlook

### ISSUE THREAD

**Issue by FlexMcMurphy** [phw198/OutlookGoogleCalendarSync#1712]
OGCS 2.10.0.0
Edition: Portable
Sync Direction: O<->G

I have a recurring appointment that I created in Evolution mail client running in Manjaro. This syncs without problem to my Google Calender but from there it will not sync to OGCS.

I tried two different patterns, each works in Evolution and syncs to Google Calendar but both fail to sync with Outlook via OGCS.
1. Starting on 6th Oct, recur Every 4 days for 10 occurrences.... the sync to OGCS fails with this in the log:
```
2023-10-02 22:34:42,018 DEBUG  31 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [306] -  Processing >> 06/10/2023 09:00 (R) => "My Test Recurring Appointment"
2023-10-02 22:34:42,018 DEBUG  31 OutlookGoogleCalendarSync.Console [294] -  06/10/2023 09:00 (R) "My Test Recurring Appointment"
2023-10-02 22:34:42,140 WARN  31 OutlookGoogleCalendarSync.Recurrence [321] -  There aren't any RRULEs present. Outlook doesn't support this: https://support.microsoft.com/en-gb/kb/2643084
2023-10-02 22:34:42,140 DEBUG  31 OutlookGoogleCalendarSync.Recurrence [321] -  rule: RRULE;X-EVOLUTION-ENDDATE=20231111T090000Z:FREQ=DAILY;COUNT=10;INTERVAL=4
2023-10-02 22:34:42,140 WARN  31 OutlookGoogleCalendarSync.Console [288] -  Appointment creation skipped: WARNING: The recurrence pattern is not compatible with Outlook. This event cannot be synced.
```
2. I also tried another.... Starting on 6th Oct, recur Every Week on Wed, Fri and Sun for 15 occurrences... the sync to OGCS fails with this in the log:
```
2023-10-02 23:00:02,154 DEBUG  29 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [306] -  Processing >> 06/10/2023 09:00 (R) => "My Test Recurring Appointment"
2023-10-02 23:00:02,154 DEBUG  29 OutlookGoogleCalendarSync.Console [294] -  06/10/2023 09:00 (R) "My Test Recurring Appointment"
2023-10-02 23:00:02,268 WARN  29 OutlookGoogleCalendarSync.Recurrence [321] -  There aren't any RRULEs present. Outlook doesn't support this: https://support.microsoft.com/en-gb/kb/2643084
2023-10-02 23:00:02,268 DEBUG  29 OutlookGoogleCalendarSync.Recurrence [321] -  rule: RRULE;X-EVOLUTION-ENDDATE=20231108T090000Z:FREQ=WEEKLY;COUNT=15;BYDAY=WE,FR,SU
2023-10-02 23:00:02,268 WARN  29 OutlookGoogleCalendarSync.Console [288] -  Appointment creation skipped: WARNING: The recurrence pattern is not compatible with Outlook. This event cannot be synced.
```

In both cases I can create that recurrence pattern using Outlooks own "Recurrence" popup dialog just fine. Despite what is said in https://support.microsoft.com/en-gb/kb/2643084 Outlook does appear to support a variety of different recurrence patterns.
Is the issue maybe something to do with timestamps or timezones? See how the log mentions --> `X-EVOLUTION-ENDDATE=20231108T090000Z`
Or is it this issue mentioned in that link?

> "The meeting contains a recurrence pattern that is defined by the RDATE property. However, the meeting does not include a corresponding RRULE property. In this case, Outlook does not open the item, even if the RDATE property specifies a supported recurrence pattern"

Can you help me debug this further and make a recurring appointment that I make in Evolution mail client appear in Outlook via OGCS?

Cheers,

Flex

**Comment by phw198** [comment]
The issue is the `RRULE;X-EVOLUTION-ENDDATE=20231108T090000Z:` part of the recurrence rule. According to the [ical standard](https://icalendar.org/iCalendar-RFC-5545/3-8-5-3-recurrence-rule.html), this is how non-standard parameters should be held - it's the first time I've ever seen it done though!

Pretty sure I just need to make OGCS ignore that bit...

**Comment by phw198** [comment]
:package: This should now be resolved with [hotfix v2.10.1.8.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/12839381/v2.10.1.8.zip) ([see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)).

Please let me know how it goes!

**Comment by FlexMcMurphy** [comment]
Hi Paul,

Yes that fix has solved my issue with recurring appointments/events.
Thanks so much for making the fix!

Flex
